subiquity/server/snapd/types.py

```diff
@@ -317,5 +318,10 @@ class SystemActionRequest:
 
 
 @snapdtype
-class SystemActionResponse:
+class SystemActionResponseSetupEncryption:
     encrypted_devices: Dict[NonExhaustive[Role], str] = attr.Factory(dict)
+
+
+@snapdtype
+class SystemActionResponseGenerateRecoveryKey:
+    recovery_key: str
```

Comment:
we might want to keep a single type and define `encrypted_devices` and `recovery_key` as `Optional` fields.

Reply:
That would work as well but I do like the clear contract here on what fields are relevant for this case.  Up to you.